fvm/evm/emulator/state/base.go

```diff
@@ -0,0 +1,472 @@
+	return false
+}
+
+// SlotInAccessList checks if a slot is in the access list
+//
+// access list control is not a functionality of the base view
+// it always returns false
+func (v *BaseView) SlotInAccessList(types.SlotAddress) (addressOk bool, slotOk bool) {
+	return false, false
+}
+
+// CreateAccount creates a new account
+func (v *BaseView) CreateAccount(
+	addr gethCommon.Address,
+	balance *big.Int,
+	nonce uint64,
+	code []byte,
+	codeHash gethCommon.Hash,
+) error {
+	var colID []byte
+	// if is an smart contract account
+	if len(code) > 0 {
+		err := v.storeCode(addr, code)
+		if err != nil {
+			return err
+		}
+
+		// create a new collection for slots
+		col, err := v.collectionProvider.NewCollection()
+		if err != nil {
+			return err
+		}
+		colID = col.CollectionID()
+	}
+
+	// create a new account and store it
+	acc := NewAccount(addr, balance, nonce, codeHash, colID)
+
+	// no need to update the cache , storeAccount would update the cache
+	return v.storeAccount(acc)
+}
+
+// UpdateAccount updates an account's meta data
+func (v *BaseView) UpdateAccount(
+	addr gethCommon.Address,
+	balance *big.Int,
+	nonce uint64,
+	code []byte,
+	codeHash gethCommon.Hash,
+) error {
+	acc, err := v.getAccount(addr)
+	if err != nil {
+		return err
+	}
+	// if update is called on a non existing account
+	// we gracefully call the create account
+	// TODO: but we might need to revisit this action in the future
+	if acc == nil {
+		return v.CreateAccount(addr, balance, nonce, code, codeHash)
+	}
+	// if it has a code change
+	if codeHash != acc.CodeHash {
+		err := v.storeCode(addr, code)
+		if err != nil {
+			return err
+		}
+		// TODO: maybe purge the state in the future as well
+		// currently the behaviour of stateDB doesn't purge the data
+	}
+	newAcc := NewAccount(addr, balance, nonce, codeHash, acc.CollectionID)
+	// no need to update the cache , storeAccount would update the cache
+	return v.storeAccount(newAcc)
+}
+
+// DeleteAccount deletes an account's meta data, code, and
+// storage slots associated with that address
+func (v *BaseView) DeleteAccount(addr gethCommon.Address) error {
+	// 1. check account exists
+	acc, err := v.getAccount(addr)
+	if err != nil {
+		return err
+	}
+	if acc == nil {
+		return fmt.Errorf("account doesn't exist to be deleted")
+	}
+
+	// 2. update the cache
+	delete(v.cachedAccounts, addr)
+
+	// 3. collections
+	err = v.accounts.Remove(addr.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// 4. remove the code
+	if acc.HasCode() {
+		err = v.storeCode(addr, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	// 5. remove storage slots
+	if len(acc.CollectionID) > 0 {
+		col, found := v.slots[addr]
+		if !found {
+			col, err = v.collectionProvider.CollectionByID(acc.CollectionID)
+			if err != nil {
+				return err
+			}
+		}
+		// delete all slots related to this account (eip-6780)
+		err = col.Destroy()
+		if err != nil {
+			return err
+		}
```

Comment:
Do we need to update `v.slots` and `v.cachedSlots` when account is destroyed?

Reply:
updated at [b5cd226](https://github.com/onflow/flow-go/pull/5166/commits/b5cd2269931f5ea70677d59199b1820dc9d2b2d4)